cmd/snap-update-ns/entry.go

```diff
@@ -98,11 +98,13 @@ func XSnapdGID(e *mount.Entry) (gid uint64, err error) {
 // XSnapdEntryID returns the identifier of a given mount enrty.
 //
 // Identifiers are kept in the x-snapd.id mount option. The value is a string
-// that identifies a mount entry and is stable across invocations of snapd (it
-// is a hash of some sort).
+// that identifies a mount entry and is stable across invocations of snapd. In
+// absence of that identifier the entry mount point is returned.
 func XSnapdEntryID(e *mount.Entry) string {
-	val, _ := e.OptStr("x-snapd.id")
-	return val
+	if val, ok := e.OptStr("x-snapd.id"); ok {
+		return val
+	}
+	return e.Dir
```

Comment:
This is something I was experimenting with. Initially it was just an opaque ID that snapd would generate. Then I realized it cannot be so because that identifier has to be stable so I resorted to hashing the whole entry. Then I realized it actually needs to be implicitly defined so that existing installations upgrade correctly (and by correctly I mean that there are no spurious changes). In the end this just became the mount point. I think that we can simplify this aspect and just use the `e.Dir` (ill-named for files but still) directly.

Reply:
"I think that we can simplify this aspect and just use the e.Dir (ill-named for files but still) directly."

Will this happen in a follow-up PR?

Reply:
I'll do it in a follow up.